src/runner.rs

```diff
@@ -320,6 +327,29 @@ impl JobRunnerOptions {
     }
 }
 
+impl JobRunnerHandle {
+    /// Return the number of still running jobs
+    pub fn num_running_jobs(&self) -> usize {
+        self.runner.running_jobs.load(Ordering::Relaxed)
+    }
+
+    /// Wait for the jobs to finish, but not more than `timeout`
+    pub async fn wait_jobs_finish(&self, timeout: Duration) {
+        let start = Instant::now();
+        let step = Duration::from_millis(1);
```

Comment:
The step doesn't need to be this small, 10ms will give other tasks more time to run.

Reply:
Other than this, the changes look great 👍